redwood/src/lib.rs

```diff
@@ -131,18 +136,27 @@ fn encrypt(
 
     // For each of the recipient certificates, pull the encryption keys that
     // are compatible with by the standard policy (e.g. not SHA-1) supported by
-    // Sequoia (duh), and not revoked.
+    // Sequoia, and not revoked.
     // These filter options should be kept in sync with `Helper::decrypt()`.
     for cert in certs.iter() {
-        for key in cert
+        let keys: Vec<ValidErasedKeyAmalgamation<PublicParts>> = cert
```

Comment:
I think Rust should be able to figure out the inner type, so in theory you should just need to write `let keys: Vec<_> = cert...collect()` (Nothing wrong with spelling out the full type if you want, I just find stuff like `ValidErasedKeyAmalgamation<PublicParts>` annoying to write so I try to skip them)